plover/steno_dictionary.py

```diff
@@ -264,11 +264,12 @@ def save(self, path_list=None):
 
         If <path_list> is None, all writable dictionaries are saved'''
         if path_list is None:
-            dict_list = [d for d in self if dictionary.save is not None]
+            dict_list = [d for d in self]
         else:
             dict_list = [self[path] for path in path_list]
         for d in dict_list:
-            d.save()
+            if not d.readonly:
```

Comment:
```suggestion
            assert not d.readonly:
```

Reply:
So it throws an error instead of silently failing to write. Got it. Will rebase on the last merge too.

Reply:
I'm not 100% sure I'm testing this code the right way, but won't an assert here crash the program? There's a second way that disc_list can be built and it doesn't look like it is checking for the readonly attribute.

Reply:
That's precisely the reason for the assert. Any code that calls this by manually specifying the paths is responsible for making sure those dictionaries aren't read-only. It's much easier to debug an AssertionError than a mysterious case of "I made changes and they didn't save."

Reply:
Ok, I just wanted to make sure there wasn't something I was missing. Like, may an assert just popped up an error window or something. An assert actually just stops Plover completely.

I'm thinking we should prevent this assert from triggering. With this change, if you try to edit a read-only dictionary, it will crash Plover.

Reply:
Ok, I just wanted to make sure there wasn't something I was missing. Like, maybe an assert just popped up an error window or something. An assert actually just stops Plover completely.

I'm thinking we should prevent this assert from triggering. With this change, if you try to edit a read-only dictionary, it will crash Plover.